diff -r 26a11f918629 cron.php
--- a/cron.php	Fri Jun 25 10:56:00 2010 -0700
+++ b/cron.php	Fri Jun 25 14:59:22 2010 -0700
@@ -6,6 +6,7 @@
  * Handles incoming requests to fire off regularly-scheduled tasks (cron jobs).
  */
 
+$cron_run = TRUE;
 include_once './includes/bootstrap.inc';
 drupal_bootstrap(DRUPAL_BOOTSTRAP_FULL);
 drupal_cron_run();
diff -r 85c83d4fadce sites/all/modules/mailq/mailq.module
--- a/sites/all/modules/mailq/mailq.module	Tue Jul 10 18:58:39 2012 -0700
+++ b/sites/all/modules/mailq/mailq.module	Tue Jul 10 18:59:37 2012 -0700
@@ -671,7 +671,7 @@
  * During a cron run, the mail will be sent.
  * Cron could be run from cron.php or from 'admin/logs/status/run-cron'
  */
-if ($_GET['q'] != 'admin/reports/status/run-cron' && (request_uri() != '/cron.php')) {
+if (!$GLOBALS['cron_run'] && $_GET['q'] != 'admin/reports/status/run-cron' && (request_uri() != '/cron.php')) {
   function drupal_mail_wrapper(&$message) {
     if (MAILQ_DEBUG_MODE) {
       watchdog(
