diff --git a/sites/all/modules/contrib/salesforce/modules/salesforce_mapping/includes/salesforce_mapping.ui_controller.inc b/sites/all/modules/contrib/salesforce/modules/salesforce_mapping/includes/salesforce_mapping.ui_controller.inc
index a68c09e..2085319 100644
--- a/sites/all/modules/contrib/salesforce/modules/salesforce_mapping/includes/salesforce_mapping.ui_controller.inc
+++ b/sites/all/modules/contrib/salesforce/modules/salesforce_mapping/includes/salesforce_mapping.ui_controller.inc
@@ -31,10 +31,6 @@ class SalesforceMappingUIController extends EntityDefaultUIController {
     $query = new EntityFieldQuery();
     $query->entityCondition('entity_type', $this->entityType);
 
-    if ($this->overviewPagerLimit) {
-      $query->pager($this->overviewPagerLimit);
-    }
-
     $results = $query->execute();
 
     $ids = isset($results[$this->entityType]) ? array_keys($results[$this->entityType]) : array();
