diff --git a/horizontal-tabs/horizontal-tabs.js b/horizontal-tabs/horizontal-tabs.js
index 977e1b6..aec4b6b 100644
--- a/horizontal-tabs/horizontal-tabs.js
+++ b/horizontal-tabs/horizontal-tabs.js
@@ -29,8 +29,10 @@ Drupal.behaviors.horizontalTabs = {
 
       // Transform each fieldset into a tab.
       $fieldsets.each(function (i) {
+        var $legend = $('> legend', this);
+        $('.element-invisible', $legend).remove();
         var horizontal_tab = new Drupal.horizontalTab({
-          title: $('> legend', this).text(),
+          title: $legend.text(),
           fieldset: $(this)
         });
         horizontal_tab.item.addClass('horizontal-tab-button-' + i);
