diff --git a/profiles/openasu/modules/contrib/radix_layouts/plugins/layouts/radix_whelan/radix-whelan.tpl.php b/profiles/openasu/modules/contrib/radix_layouts/plugins/layouts/radix_whelan/radix-whelan.tpl.php
index c64f960..724d37c 100755
--- a/profiles/openasu/modules/contrib/radix_layouts/plugins/layouts/radix_whelan/radix-whelan.tpl.php
+++ b/profiles/openasu/modules/contrib/radix_layouts/plugins/layouts/radix_whelan/radix-whelan.tpl.php
@@ -11,25 +11,25 @@
 ?>
 
 <div class="panel-display whelan clearfix <?php if (!empty($classes)) { print $classes; } ?><?php if (!empty($class)) { print $class; } ?>" <?php if (!empty($css_id)) { print "id=\"$css_id\""; } ?>>
-
-<div class="container-fluid">
-  <div class="row">
-    <div class="col-md-3 radix-layouts-column1 panel-panel">
-      <div class="panel-panel-inner">
-        <?php print $content['column1']; ?>
-      </div>
-    </div>
-    <div class="col-md-6 radix-layouts-content panel-panel">
-      <div class="panel-panel-inner">
-        <?php print $content['contentmain']; ?>
-      </div>
-    </div>
-    <div class="col-md-3 radix-layouts-column2 panel-panel">
-      <div class="panel-panel-inner">
-        <?php print $content['column2']; ?>
+  <section class="section">
+    <div class="container">
+      <div class="row">
+        <div class="col-md-<?php print kalatheme_grid_size(KALATHEME_GRID_FOURTH, 3); ?> whelan-column-content-region-1 radix-layouts-column1 panel-panel">
+          <div class="panel-panel-inner">
+            <?php print $content['column1']; ?>
+          </div>
+        </div>
+        <div class="col-md-<?php print kalatheme_grid_size(KALATHEME_GRID_HALF, 3); ?> whelan-column-content-region-2 radix-layouts-content panel-panel">
+          <div class="panel-panel-inner">
+            <?php print $content['contentmain']; ?>
+          </div>
+        </div>
+        <div class="col-md-<?php print kalatheme_grid_size(KALATHEME_GRID_FOURTH, 3); ?> whelan-column-content-region-3 radix-layouts-column2 panel-panel">
+          <div class="panel-panel-inner">
+            <?php print $content['column2']; ?>
+          </div>
+        </div>
       </div>
     </div>
-  </div>
-</div>
-  
+  </section>
 </div><!-- /.whelan -->
diff --git a/profiles/openasu/modules/webspark_featurescustom/webspark_banner/templates/fieldable-panels-pane--banner.tpl.php b/profiles/openasu/modules/webspark_featurescustom/webspark_banner/templates/fieldable-panels-pane--banner.tpl.php
index 0e331b5..dc27ec1 100644
--- a/profiles/openasu/modules/webspark_featurescustom/webspark_banner/templates/fieldable-panels-pane--banner.tpl.php
+++ b/profiles/openasu/modules/webspark_featurescustom/webspark_banner/templates/fieldable-panels-pane--banner.tpl.php
@@ -1,5 +1,6 @@
 <div class="<?php print $classes; ?>"<?php print $attributes; ?>>
 
+  <?php dpm(get_defined_vars()); ?>
 	<?php print render($elements['title']); ?>
   <?php print render($title_suffix); ?>
   <?php print $fields; ?>
