commit 5b9a93e5f8ca9c71fad7b2d1f3b15541bcd79812
Author: RichardNesbitt <richardknesbitt@gmail.com>
Date:   Tue Sep 18 14:15:13 2018 -0700

    Menu patch test

diff --git a/profiles/openasu/modules/contrib/tb_megamenu/templates/tb-megamenu-item.tpl.php b/profiles/openasu/modules/contrib/tb_megamenu/templates/tb-megamenu-item.tpl.php
index 79e255957..050c9f345 100644
--- a/profiles/openasu/modules/contrib/tb_megamenu/templates/tb-megamenu-item.tpl.php
+++ b/profiles/openasu/modules/contrib/tb_megamenu/templates/tb-megamenu-item.tpl.php
@@ -1,9 +1,7 @@
 <li <?php print $attributes;?> class="<?php print $classes;?>" role="listitem" aria-level="<?php print $level; ?>">
-  <a href="<?php print in_array($item['link']['href'], array('<nolink>')) ? "#" : url($item['link']['href']);?>"
-     class="<?php print implode(" ", $a_classes);?>"
-     <?php if ($submenu): ?>aria-haspopup="true"<?php endif; ?>
-  >    <?php if(!empty($item_config['xicon'])) : ?>
-      <i class="<?php print $item_config['xicon'];?>"></i>
+  <a href="<?php print in_array($item['link']['href'], array('<nolink>')) ? "#" : url($item['link']['href'], $item['link']['options']);?>" <?php echo drupal_attributes($item['link']['#attributes']); ?><?php if ($submenu): ?> aria-haspopup="true"<?php endif; ?>>
+    <?php if(!empty($item_config['xicon'])) : ?>
+      <span class="<?php print $item_config['xicon'];?>"></span>
     <?php endif;?>    
     <?php print t($item['link']['title']);?>
     <?php if($submenu && $block_config['auto-arrow']) :?>
