diff --git a/modules/media_wysiwyg/includes/media_wysiwyg.filter.inc b/modules/media_wysiwyg/includes/media_wysiwyg.filter.inc
index 0a16b87..57211f9 100644
--- a/modules/media_wysiwyg/includes/media_wysiwyg.filter.inc
+++ b/modules/media_wysiwyg/includes/media_wysiwyg.filter.inc
@@ -302,6 +302,11 @@ function _media_wysiwyg_generate_tagMap($text) {
   static $tagmap = array();
   preg_match_all("/\[\[.*?\]\]/s", $text, $matches, PREG_SET_ORDER);
   foreach ($matches as $match) {
+    // Fix macros that may have gotten HTML-encoded due to this bug:
+    // https://www.drupal.org/node/2028253
+    if(strpos($match[0], '[[{&quot;') === 0) {
+      $match[0] = htmlspecialchars_decode($match[0]);
+    }
     // We see if tagContent is already in $tagMap, if not we add it
     // to $tagmap.  If we return an empty array, we break embeddings of the same
     // media multiple times.
