diff --git a/profiles/openasu/modules/contrib/backgroundfield/backgroundfield.module b/profiles/openasu/modules/contrib/backgroundfield/backgroundfield.module
index c251a8a..dc153db 100644
--- a/profiles/openasu/modules/contrib/backgroundfield/backgroundfield.module
+++ b/profiles/openasu/modules/contrib/backgroundfield/backgroundfield.module
@@ -178,6 +178,14 @@ function backgroundfield_field_delete_revision($entity_type, $entity, $field, $i
 function backgroundfield_field_update_field($field, $prior_field, $has_data) {
   // hack so we can use the default image function
   $field['type'] = 'image';
+  // WEBSPARK-533 - Patch based on https://www.drupal.org/files/backgroundfield_undefined_index-1841978-18.patch
+  // to suppress default_image error.
+  if (!isset($field['settings']['default_image'])) {
+    $field['settings']['default_image'] = NULL;
+  }
+  if (!isset($prior_field['settings']['default_image'])) {
+    $prior_field['settings']['default_image'] = NULL;
+  }
   image_field_update_field($field, $prior_field, $has_data);
 }
 
