diff --git a/profiles/openasu/modules/panopoly/panopoly_test/.gitignore b/profiles/openasu/modules/panopoly/panopoly_test/.gitignore
index ec14d1e..3eb013e 100644
--- a/profiles/openasu/modules/panopoly/panopoly_test/.gitignore
+++ b/profiles/openasu/modules/panopoly/panopoly_test/.gitignore
@@ -2,3 +2,5 @@ behat.yml
 composer.phar
 bin
 vendor
+tests/behat.local.yml
+
diff --git a/profiles/openasu/modules/panopoly/panopoly_test/tests/behat.common.yml b/profiles/openasu/modules/panopoly/panopoly_test/tests/behat.common.yml
index c22bdf9..56eceae 100644
--- a/profiles/openasu/modules/panopoly/panopoly_test/tests/behat.common.yml
+++ b/profiles/openasu/modules/panopoly/panopoly_test/tests/behat.common.yml
@@ -18,7 +18,7 @@ default:
       goutte: ~
       selenium2: ~
       #base_url: http://localhost
-      #files_path: './tests'
+      #files_path: './profiles/openasu/modules/panopoly/panopoly_test/tests'
     Drupal\DrupalExtension:
       blackbox: ~
       api_driver: 'drupal'
diff --git a/profiles/openasu/modules/panopoly/panopoly_test/tests/behat.local.yml b/profiles/openasu/modules/panopoly/panopoly_test/tests/behat.local.yml
index 50cf29f..a44bb87 100644
--- a/profiles/openasu/modules/panopoly/panopoly_test/tests/behat.local.yml
+++ b/profiles/openasu/modules/panopoly/panopoly_test/tests/behat.local.yml
@@ -7,7 +7,7 @@ default:
       goutte: ~
       selenium2: ~
       base_url: http://bjdevil21-webspark
-      files_path: '/home/bryan/github.com/bjdevil21/webspark-drops-drupal7/tests'
+      files_path: '/home/bryan/github.com/bjdevil21/webspark-drops-drupal7/profiles/openasu/modules/panopoly/panopoly_test/tests'
     Drupal\DrupalExtension:
       blackbox: ~
       drush:
diff --git a/profiles/openasu/modules/panopoly/panopoly_test/tests/behat.travis.yml b/profiles/openasu/modules/panopoly/panopoly_test/tests/behat.travis.yml
index 485e22b..9d9568a 100644
--- a/profiles/openasu/modules/panopoly/panopoly_test/tests/behat.travis.yml
+++ b/profiles/openasu/modules/panopoly/panopoly_test/tests/behat.travis.yml
@@ -5,7 +5,7 @@ default:
   extensions:
     Behat\MinkExtension:
       base_url: http://127.0.0.1:8888
-      files_path: './tests'
+      files_path: './profiles/openasu/modules/panopoly/panopoly_test/tests'
     Drupal\DrupalExtension:
       drush:
         root: "/home/travis/build/ASU/webspark-drops-drupal7"
diff --git a/profiles/openasu/modules/panopoly/panopoly_test/tests/features/basic_ipe.feature b/profiles/openasu/modules/panopoly/panopoly_test/tests/features/basic_ipe.feature
index 909890c..7c3d6a4 100644
--- a/profiles/openasu/modules/panopoly/panopoly_test/tests/features/basic_ipe.feature
+++ b/profiles/openasu/modules/panopoly/panopoly_test/tests/features/basic_ipe.feature
@@ -6,7 +6,13 @@ Feature: Make basic changes with the IPE
   Background:
     Given I am logged in as a user with the "administrator" role
       And Panopoly magic live previews are disabled
-      And I am viewing a "panopoly_test_page" with the title "Basic IPE Testing title"
+      When I visit "/node/add/panopoly-test-page"
+      And I fill in the following:
+        | Title  | Basic IPE Testing title |
+        | Editor | plain_text    |
+        | Body   | Basic IPE Testing body  |
+      And I press "edit-submit"
+      Then I should see "Basic IPE Testing title"
 
   @api @javascript @webspark_broken @webspark_fixed
   Scenario: Change layout
diff --git a/profiles/openasu/modules/panopoly/panopoly_test/tests/features/wysiwyg.feature b/profiles/openasu/modules/panopoly/panopoly_test/tests/features/wysiwyg.feature
index 2e848ec..7dd372e 100644
--- a/profiles/openasu/modules/panopoly/panopoly_test/tests/features/wysiwyg.feature
+++ b/profiles/openasu/modules/panopoly/panopoly_test/tests/features/wysiwyg.feature
@@ -39,10 +39,10 @@ Feature: Use rich text editor
       | Align Center                  | p          | text-align      | center       |
       | Align Right                   | p          | text-align      | right        |
 
-  @api @javascript @panopoly_wysiwyg
+  # Removed actions related to buttons that were removed by Webspark WYSIWYG
+  @api @javascript @panopoly_wysiwyg @webspark_broken @webspark_fixed
   Scenario Outline: Format text in the editor (advanced toolbar)
-    When I expand the toolbar in the "edit-body-und-0-value" WYSIWYG editor
-      And I click the "<Action>" button in the "edit-body-und-0-value" WYSIWYG editor
+    When I click the "<Action>" button in the "edit-body-und-0-value" WYSIWYG editor
       And I type "Testing body" in the "edit-body-und-0-value" WYSIWYG editor
       #And I press "Publish"
       And I press "edit-submit"
@@ -50,6 +50,5 @@ Feature: Use rich text editor
 
     Examples:
       | Action          | Element | Property        | Value     |
-      | Underline       | span    | text-decoration | underline |
       | Align Full      | p       | text-align      | justify   |
       | Increase Indent | p       | padding-left    | 30px      |
