diff --git a/profiles/openasu/modules/contrib/tb_megamenu/js/tb-megamenu-object.js b/profiles/openasu/modules/contrib/tb_megamenu/js/tb-megamenu-object.js
index 4eaa01f..7f9d6e3 100644
--- a/profiles/openasu/modules/contrib/tb_megamenu/js/tb-megamenu-object.js
+++ b/profiles/openasu/modules/contrib/tb_megamenu/js/tb-megamenu-object.js
@@ -121,7 +121,7 @@ Drupal.TBMegaMenu = Drupal.TBMegaMenu || {};
     }
     else {
       unbindEvents(sub);
-      if (liitem.find('ul.level'+liitem.attr('data-level')).length > 0) {
+      if (liitem.find('ul.level-'+liitem.attr('data-level')).length > 0) {
         sub.css('display','none');
         liitem.attr('data-hidesub', 1);
       }
