diff --git a/date_elements.inc b/date_elements.inc
index b306e9d8..99633190 100644
--- a/date_elements.inc
+++ b/date_elements.inc
@@ -102,7 +102,7 @@ function date_field_widget_form(&$form, &$form_state, $field, $instance, $langco
     '#type' => 'date_combo',
     '#theme_wrappers' => array('date_combo'),
     '#weight' => $delta,
-    '#default_value' => isset($items[$delta]) ? $items[$delta] : '',
+    '#default_value' => isset($items[$delta]) ? $items[$delta] : array(),
     '#date_timezone' => $timezone,
     '#element_validate' => array('date_combo_validate'),
     '#date_is_default' => $is_default,
