diff --git a/components/email.inc b/components/email.inc
index bbabf6966..7bddee96f 100644
--- a/components/email.inc
+++ b/components/email.inc
@@ -234,7 +234,14 @@ function theme_webform_email($variables) {
       $element['#attributes'][$property] = $element['#' . $property];
     }
   }
-  _form_set_class($element, array('form-text', 'form-email'));
+
+  // Add form-control class if display is not inline
+  $email_classes = array('form-text', 'form-email');
+  if ($element["#title_display"] !== "inline") {
+    $email_classes[] = 'form-control';
+  }
+
+  _form_set_class($element, $email_classes);
 
   return '<input' . drupal_attributes($element['#attributes']) . ' />';
 }
diff --git a/components/number.inc b/components/number.inc
index c2bf76e72..091d4cba1 100644
--- a/components/number.inc
+++ b/components/number.inc
@@ -275,7 +275,14 @@ function theme_webform_number($variables) {
       $element['#attributes'][$property] = $element['#' . $property];
     }
   }
-  _form_set_class($element, array('form-text', 'form-number'));
+
+  // Add form-control class if display is not inline
+  $number_classes = array('form-text', 'form-number');
+  if ($element['#title_display'] !== 'inline') {
+    $number_classes[] = 'form-control';
+  }
+
+  _form_set_class($element, $number_classes);
 
   return '<input' . drupal_attributes($element['#attributes']) . ' />';
 }
