diff --git a/includes/entity.inc b/includes/entity.inc
index e80ce3b89f..b3cd1a62aa 100644
--- a/includes/entity.inc
+++ b/includes/entity.inc
@@ -172,7 +172,7 @@ class DrupalDefaultEntityController implements DrupalEntityControllerInterface {
     // were passed. The $ids array is reduced as items are loaded from cache,
     // and we need to know if it's empty for this reason to avoid querying the
     // database when all requested entities are loaded from cache.
-    $passed_ids = !empty($ids) ? array_flip($ids) : FALSE;
+    $passed_ids = is_array($ids) && !empty($ids) && !is_array(current($ids)) ? array_flip(array_filter($ids)) : FALSE;
     // Try to load entities from the static cache, if the entity type supports
     // static caching.
     if ($this->cache && !$revision_id) {
@@ -385,7 +385,8 @@ class DrupalDefaultEntityController implements DrupalEntityControllerInterface {
     // Load any available entities from the internal cache.
     if (!empty($this->entityCache)) {
       if ($ids) {
-        $entities += array_intersect_key($this->entityCache, array_flip($ids));
+        if (is_array(current($ids))) { $ids = current($ids); }
+        $entities += array_intersect_key($this->entityCache, array_flip(array_filter((array) $ids)));
       }
       // If loading entities only by conditions, fetch all available entities
       // from the cache. Entities which don't match are removed later.
